### ISSUE TITLE
Unexpected error typing Python function definition

### ISSUE THREAD

**Issue by jkloth** [Microsoft/PTVS#2547]
While typing the following:
```
def test(obj,
         obj,
```
[crashes on the last comma]
Note the *lack* of trailing paren and multiple lines.

[Window Title]
devenv.exe

[Main Instruction]
An unexpected error occurred

[Content]
Please press Ctrl+C to copy the contents of this dialog and report this error to our issue tracker.

[^] Hide details  [Close]

[Expanded Information]
```
Build: 3.0.17114.1

System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values.
Parameter name: length
   at Microsoft.VisualStudio.Text.Span..ctor(Int32 start, Int32 length)
   at Microsoft.PythonTools.Intellisense.LocationTracker.TranslateForward(Span from)
   at Microsoft.PythonTools.Intellisense.LocationTracker.TranslateForward(Int32 position)
   at Microsoft.PythonTools.Intellisense.VsProjectAnalyzer.<ConvertOutliningTags>d__160.MoveNext()
   at System.Linq.Enumerable.<ConcatIterator>d__58`1.MoveNext()
   at System.Linq.Enumerable.<ConcatIterator>d__58`1.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Microsoft.PythonTools.OutliningTaggerProvider.OutliningTagger.<>c__DisplayClass14_1.<UpdateTagsAsync>b__0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.PythonTools.OutliningTaggerProvider.OutliningTagger.<UpdateTagsAsync>d__14.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudioTools.UIThread.<InvokeTaskHelper>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.PythonTools.Infrastructure.VSTaskExtensions.<HandleAllExceptions>d__5.MoveNext()
```

**Comment by huguesv** [comment]
On a debug build, I got this assert while trying to repro:
```
---------------------------
Assertion Failed: Abort=Quit, Retry=Debug, Ignore=Continue
---------------------------
from span '[18..18)' was longer than the text snapshot



   at Microsoft.PythonTools.Intellisense.LocationTracker.TranslateForward(Span from) in C:\Projects\huguesv\PTVS\Python\Product\PythonTools\PythonTools\Intellisense\LocationTracker.cs:line 73

   at Microsoft.PythonTools.Intellisense.LocationTracker.TranslateForward(Int32 position) in C:\Projects\huguesv\PTVS\Python\Product\PythonTools\PythonTools\Intellisense\LocationTracker.cs:line 100

   at Microsoft.PythonTools.Intellisense.VsProjectAnalyzer.<ConvertOutliningTags>d__160.MoveNext() in C:\Projects\huguesv\PTVS\Python\Product\PythonTools\PythonTools\Intellisense\ProjectAnalyzer.cs:line 2148

   at System.Linq.Enumerable.<ConcatIterator>d__58`1.MoveNext()

   at System.Linq.Enumerable.<ConcatIterator>d__58`1.MoveNext()

   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)

   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)

   at Microsoft.PythonTools.OutliningTaggerProvider.OutliningTagger.<>c__DisplayClass14_1.<UpdateTagsAsync>b__0() in C:\Projects\huguesv\PTVS\Python\Product\PythonTools\PythonTools\OutliningTaggerProvider.cs:line 119

   at System.Threading.Tasks.Task`1.InnerInvoke()

   at System.Threading.Tasks.Task.Execute()

   at System.Threading.Tasks.Task.ExecutionContextCallback(Object obj)

   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)

   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)

   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)

   at System.Threading.Tasks.Task.ExecuteEntry(Boolean bPreventDoubleExecution)

   at System.Threading.Tasks.Task.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()

   at System.Threading.ThreadPoolWorkQueue.Dispatch()

   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()


---------------------------
Abort   Retry   Ignore   
---------------------------

```

**Comment by zooba** [comment]
@huguesv @int19h I'm declaring this a release blocking bug, since so many people are hitting it. Feel free to self-assign and take a look if you like

**Comment by int19h** [comment]
Looks like this is a dupe of #2479, which @zooba has already fixed - same stack, and it already has `catch (ArgumentOutOfRangeException)` in the spot where this is supposed to be happening.

**Comment by huguesv** [comment]
Yes it is a duplicate, but I reproduced the assert AFTER the other bug was fixed, so there is still some issue to resolve here.  I just tried it again with today's code, and I got the assert on first try.

**Comment by int19h** [comment]
I'm not sure the asserts are actually meaningful. The assert is on the code path that catches (and swallows) the exception. @zooba, do you remember why you kept them in there after adding the `catch` in the fix for #2479? Was it for some later investigation into the underlying root cause?

In any case, I think this shouldn't be considered release blocking now, since it's not a crash anymore.

**Comment by zooba** [comment]
You're right - this is fixed and just not released yet.
